torchgen/gen.py

```diff
@@ -1623,8 +1623,11 @@ def get_native_function_schema_registrations(
     schema_selector: SelectiveBuilder,
 ) -> Tuple[List[str], str]:
     ns_native_functions: Dict[str, List[NativeFunction]] = defaultdict(list)
+    # a map from namespace to whether we should use TORCH_LIBRARY or TORCH_LIBRARY_FRAGMENT
+    ns_to_is_fragment: Dict[str, bool] = defaultdict(bool)
     for native_function in native_functions:
         ns_native_functions[native_function.namespace].append(native_function)
+        ns_to_is_fragment[native_function.namespace] |= "fragment" in native_function.tags
```

Comment:
Add it here: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/tags.yaml